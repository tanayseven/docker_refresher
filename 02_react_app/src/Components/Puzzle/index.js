import React, { Component } from 'react';
import Answer from './../Answer';
import PropTypes from 'prop-types';
import './Puzzle.css';

class Puzzle extends Component {
  render() {
    const { data, handleClick } = this.props
    return (
      <div className="Puzzle">
        <h1 className="Puzzle__Statement">
          {data.question}
        </h1>
        <div className="Puzzle__Answers">
          {data.answers.map((answer, i) => {
            return (
              <Answer
                currentIndex={i + 1}
                key={`${answer}-${i}`}
                value={answer}
                handleClick={handleClick}
              />
            )
          })}
        </div>
      </div>
    )
  }
}

Puzzle.propTypes = {
  data: PropTypes.shape({
    question: PropTypes.string.isRequired,
    answers: PropTypes.arrayOf(PropTypes.number).isRequired
  }).isRequired,
  handleClick: PropTypes.func.isRequired
}

export default Puzzle;