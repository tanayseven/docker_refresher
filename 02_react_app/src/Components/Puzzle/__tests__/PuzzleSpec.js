import React from 'react';
import renderer from 'react-test-renderer';
import Puzzle from './../index';

describe('Puzzle', () => {
    it('renders correctly', () => {
        const puzzleProps = {
            data: {
                question: "Sample question ?",
                answers: [1, 5, 8, 9]
            },
            handleClick: jest.fn()
        }
        let tree = renderer.create(
            <Puzzle {...puzzleProps}/>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});