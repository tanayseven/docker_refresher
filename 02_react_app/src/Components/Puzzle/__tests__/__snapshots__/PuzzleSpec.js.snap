// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puzzle renders correctly 1`] = `
<div
  className="Puzzle"
>
  <h1
    className="Puzzle__Statement"
  >
    Sample question ?
  </h1>
  <div
    className="Puzzle__Answers"
  >
    <div
      className="Answer"
    >
      <button
        className="Answer--Btn"
        onClick={[Function]}
        onTouchStart={[Function]}
      >
        <span>
          1
          ) 
        </span>
         
        1
      </button>
    </div>
    <div
      className="Answer"
    >
      <button
        className="Answer--Btn"
        onClick={[Function]}
        onTouchStart={[Function]}
      >
        <span>
          2
          ) 
        </span>
         
        5
      </button>
    </div>
    <div
      className="Answer"
    >
      <button
        className="Answer--Btn"
        onClick={[Function]}
        onTouchStart={[Function]}
      >
        <span>
          3
          ) 
        </span>
         
        8
      </button>
    </div>
    <div
      className="Answer"
    >
      <button
        className="Answer--Btn"
        onClick={[Function]}
        onTouchStart={[Function]}
      >
        <span>
          4
          ) 
        </span>
         
        9
      </button>
    </div>
  </div>
</div>
`;
